### PR TITLE
Move skillmap projects to the editor workspace

### PR DESCRIPTION
**Pull request by riknoll**
Moving all of the skillmap projects from outside the editor iframe to inside! List of changes includes:

1. Skillmap headers are now saved along all of the other editor projects with a flag to filter them out of "my projects"
2. When we load the workspace, skillmap projects are automatically imported and soft-deleted from the skillmap workspace
3. All code carryover project merging now happens inside the iframe when the tutorial is loaded
4. Created some new tutorial events for reporting progress from the editor to the MakecodeFrame
5. The skillmap now has the webconfig embedded in the HTML file and points to the correct target version instead of the embedURL for the iframe
6. The "start fresh"/"keep code" dialog now shows *before* we load the editor instead of *after*

Here's a test upload:

* [editor](https://arcade.makecode.com/app/e0f5b5912347d2ab2265fcc9ae4b863f8c77451a-e68d4ba962)
* [skillmap](https://arcade.makecode.com/app/e0f5b5912347d2ab2265fcc9ae4b863f8c77451a-e68d4ba962---skillmap)

**Note**: If you open this editor/skillmap upload, all of your skillmap tutorial projects will suddenly show up in the beta/released editor. The code to filter skillmap projects out of "My Projects" is part of this PR

TODO:
1. Now that the dialog is before the editor load, we should start loading the editor in the background to improve perf
2. I need to fix the loading editor event so that you don't ever see "starting tutorial" when opening an activity

### CODE REVIEW

File: skillmap/src/actions/types.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const CLOSE_ACTIVITY = "CLOSE_ACTIVITY";
 export const SAVE_AND_CLOSE_ACTIVITY = "SAVE_AND_CLOSE_ACTIVITY";
 export const RESTART_ACTIVITY = "RESTART_ACTIVITY";
 export const SET_RELOAD_HEADER_STATE = "SET_RELOAD_HEADER_STATE";
+export const SET_CARRYOVER_CODE_STATE = "SET_CARRYOVER_CODE_STATE";
```
Comment (shakao) [inline skillmap/src/actions/types.ts:13]: needed?

File: skillmap/src/components/AppModal.tsx [hunk 2]
```diff
@@ -122,7 +123,8 @@ export class AppModalImpl extends React.Component<AppModalProps, AppModalState>
             { label: lf("CANCEL"), onClick: this.handleOnClose },
             { label: lf("RESTART"), onClick: () => {
                 tickEvent("skillmap.activity.restart", { path: mapId, activity: activity!.activityId });
-                dispatchRestartActivity(mapId, activity!.activityId);
+                if (showCodeCarryoverModal) dispatchShowCarryoverModal(mapId, activity!.activityId);
```
Comment (shakao) [inline skillmap/src/components/AppModal.tsx:126]: is it weird to have two subsequent modals like this? do the modals look different enough that kids won't just rapidly click on the rightmost button twice in a row?

File: webapp/src/projects.tsx [hunk 3]
```diff
@@ -1196,7 +1206,14 @@ function applyCodeCardAction(projectView: pxt.editor.IProjectView, ticSrc: "proj
     }
 
     function chgCode(name: string, path: string, loadBlocks: boolean, preferredEditor?: string, template?: pxt.ProjectTemplate) {
-        return projectView.startActivity("example", path, name, preferredEditor, null, { path, name, loadBlocks, prj: template });
+        // return projectView.startActivity("example", path, name, preferredEditor, null, );
```
Comment (shakao) [inline webapp/src/projects.tsx:1209]: needed?

File: skillmap/src/components/makecodeFrame.tsx [hunk 4]
```diff
@@ -20,62 +20,69 @@ interface MakeCodeFrameProps {
...
-    carryoverModalVisible: boolean;
+    carryoverCode: boolean;
+    previousHeaderId?: string;
+    progress?: ActivityState;
     dispatchSetHeaderIdForActivity: (mapId: string, activityId: string, id: string, currentStep: number, maxSteps: number, isCompleted: boolean) => void;
     dispatchCloseActivity: (finished?: boolean) => void;
     dispatchSaveAndCloseActivity: () => void;
     dispatchUpdateUserCompletedTags: () => void;
-    dispatchShowCarryoverModal: (mapId: string, activityId: string) => void;
     dispatchSetReloadHeaderState: (state: "reload" | "reloading" | "active") => void;
```
Comment (shakao) [inline skillmap/src/components/makecodeFrame.tsx:32]: do we still need any of the reloadHeaderState flow now that the carryover code has moved?

File: webapp/src/app.tsx [hunk 5]
```diff
@@ -3859,6 +3978,10 @@ export class ProjectView
         return this.state.tutorialOptions != undefined;
     }
 
+    onTutorialLoaded() {
```
Comment (shakao) [inline webapp/src/app.tsx:3981]: "tutorial.editorLoaded" tick event got dropped?

File: webapp/src/projects.tsx (hunk 3 above)
Comment (jwunderl) [inline webapp/src/projects.tsx:1209]:
```suggestion
```

**Review by jwunderl — APPROVED** [review]

File: webapp/src/app.tsx (hunk 5 above)
Comment (riknoll) [reply webapp/src/app.tsx:3981]: I figured that as long as I was piping tutorial events I might as well make it explicit and add a new event

Comment (shakao) [reply webapp/src/app.tsx:3981]: oh i mean, do we still want it for telemetry reasons?

Comment (riknoll) [reply webapp/src/app.tsx:3981]: oh yeah sure i'll put it back

**Comment by shakao** [comment]
hmm the completion modal/confetti autodisplay doesn't seem to be happening in the builds here

File: skillmap/src/components/makecodeFrame.tsx (hunk 4 above)
Comment (riknoll) [reply skillmap/src/components/makecodeFrame.tsx]: Nope! removed

File: skillmap/src/actions/types.ts (hunk 1 above)
Comment (riknoll) [reply skillmap/src/actions/types.ts]: nope! removed

**Review by shakao — APPROVED** [review]

File: skillmap/src/components/AppModal.tsx (hunk 2 above)
Comment (riknoll) [reply skillmap/src/components/AppModal.tsx:125]: I think we should maybe take a look at it in a design meeting; could be confusing.

**Pull request merged**